src/patches/Node.js

```diff
@@ -262,6 +261,12 @@ export const NodePatches = utils.getOwnPropertyDescriptors({
     if (ref_node === node) {
       return node;
     }
+    scheduleObserver(this, node);
+    // optimization: assume native insertBefore is ok if the nodes are not in the document.
+    if (this.ownerDocument !== doc && node.ownerDocument !== doc) {
```

Comment:
Let's move this back to the top of the function. This is a fast path for non-main document nodes where we assume there is no shadowRoot. In that case the observer is not needed.

Reply:
Why is that?

Reply:
Oh, I guess because entering that conditional would already prevent the observer from triggering?